### PR TITLE
Fix #46: Add loading controls to tree page.

### LINKED ISSUE #46: Move (or duplicate) the "loading" controls to the main tree view
Instead of having to open a notebook to load one from a URL or from github, it should be possible to directly open one from the tree view.

### PR DESCRIPTION
**Pull request by mdboom**
This adds the loading-related buttons to the tree page, but otherwise is 
the same code and functionality.

It reorganizes the code into a notebook extension, a tree extension (the
file browser page), and code that is common between the two.

Since extension files have been added, this requires uninstalling and
reinstalling the notebook extensions (as per our README.md).

### CODE REVIEW

**Comment by jezdez** [comment]
Woah, I'll be on PTO tomorrow so this will need to wait for a proper review till next week. Nice work from what I've seen!

**Comment by mdboom** [comment]
No worries, @jezdez.  Enjoy your PTO!

File: src/jupyter_notebook_gist/static/common.js [hunk 1]
```diff
@@ -134,8 +124,8 @@ define([
                 }
                 download_nb_on_server(url, newname, true);
             } else if (xhr.status == 200) {
-                window.open(url_path_split(Jupyter.notebook.notebook_path)[0] + \
-                            encodeURIComponent(this.responseText));
+                window.open(
+                    '/notebooks/' + encodeURIComponent(xhr.responseText));
```
Comment (jezdez) [inline src/jupyter_notebook_gist/static/common.js]: Is there a reason to hardcode the path here?

Comment (mdboom) [reply src/jupyter_notebook_gist/static/common.js]: I'll see if I can get this programmatically somehow.  The difference is that there isn't a clear way to get this from the tree page -- but maybe it's there if I poke around a bit more.  I also learned from https://github.com/mozilla/jupyter-spark/pull/33 that this is missing the (optional) base_url.

Comment (mdboom) [reply src/jupyter_notebook_gist/static/common.js]: Hmm... It's hardcoded in Jupyter notebook too: https://github.com/jupyter/notebook/blob/master/notebook/static/notebook/js/notebook.js#L2947

I think the best I can do here is fix the problem with base_url, but "notebooks" will still be hard-coded.

Comment (jezdez) [reply src/jupyter_notebook_gist/static/common.js]: Good digging, that works for me.

Comment (mdboom) [reply src/jupyter_notebook_gist/static/common.js]: I've improved this, and fixed various issues that were present when a `base_url` is set.

**Pull request merged**